File: src/models/SendMessageV2.ts
class SendMessageV2Request {
  public base64_attachments: Array<string>;
  public message: string;
  public number: string;
  public recipients: Array<string>;

  constructor(base64_attachments: Array<string>, message: string, number: string, recipients: Array<string>) {
    this.base64_attachments = base64_attachments;
    this.message = message;
    this.number = number;
    this.recipients = recipients;
  }
}

interface SendMessageV2Response {
  timestamp: string;
}

export { SendMessageV2Request, SendMessageV2Response };

File: mock_server/src/models/send_message_v2.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize)]
pub struct SendMessageV2Request {
    pub base64_attachments: Vec<String>,
    pub message: String,
    pub number: String,
    pub recipients: Vec<String>,
}

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SendMessageV2Response {
    pub timestamp: String,
}
